packages/library-base/src/commonMain/kotlin/io/realm/internal/LiveRealm.kt

```diff
@@ -39,8 +39,8 @@ import kotlinx.coroutines.CoroutineDispatcher
  */
 internal abstract class LiveRealm(val owner: RealmImpl, configuration: InternalConfiguration, dispatcher: CoroutineDispatcher? = null) : BaseRealmImpl(configuration) {
 
-    private val realmChangeRegistration: RegistrationToken
-    private val schemaChangeRegistration: RegistrationToken
+    private val realmChangeRegistration: RealmCallbackTokenPointer
```

Comment:
The `RealmCallbackTokenPointer` should probably somehow be wrapped in a `NotificationToken` so we are able to both `cancel` them when the realm is closed but also release it if the objects are GC'ed when the realm itself is GC'ed. As it is right here we still cannot eliminate the explicit [`unregisterCallback`](https://github.com/realm/realm-kotlin/pull/596/files#diff-4d0528b8584520f9db14ff4031d83ca6cb837bcd889022c4abcc08b6309235baR188) to make our `garbageCollectorShouldFreeNativeResources` memory test pass.